apps/devhub/tests/test_views.py

```diff
@@ -2997,6 +3004,12 @@ def test_unique_name(self):
         eq_(r.context['new_addon_form'].non_field_errors(),
             ['This name is already in use. Please choose another.'])
 
+    def test_unlisted_name_not_unique(self):
+        """We don't enforce name uniqueness for unlisted add-ons."""
+        addon = addon_factory(name='xpi name')
+        addon.update(is_listed=False)
+        self.post()
```

Comment:
Can you add a comment? It would also be nice if the listed test used the same pattern, but with `self.assertRaises` just so we know the alternate case works.